File: backend/Billing.Api/Entities/InvoiceItem.cs
namespace Billing.Api.Entities;

public class InvoiceItem
{
    public Guid Id { get; set; }

    public Guid InvoiceId { get; set; }

    public Guid ProductId { get; set; }

    public string ProductCode { get; set; } = string.Empty;

    public string ProductDescription { get; set; } = string.Empty;

    public int Quantity { get; set; }
}

File: frontend/src/app/core/models/invoice-item.ts
export interface InvoiceItem {
  id: string;
  invoiceId: string;
  productId: string;
  productCode: string;
  productDescription: string;
  quantity: number;
}

export interface AddInvoiceItemRequest {
  productId: string;
  quantity: number;
}

export interface UpdateInvoiceItemRequest {
  quantity: number;
}
